File: opencode/aspnet/rawdog/LooseNotes/Models/Note.cs
using System.ComponentModel.DataAnnotations;

namespace LooseNotes.Models;

public class Note
{
    public int Id { get; set; }
    
    [Required]
    [StringLength(200)]
    public string Title { get; set; } = string.Empty;
    
    [Required]
    public string Content { get; set; } = string.Empty;
    
    public bool IsPublic { get; set; } = false;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    
    public string UserId { get; set; } = string.Empty;
    public ApplicationUser User { get; set; } = null!;
    
    public ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
    public ICollection<Rating> Ratings { get; set; } = new List<Rating>();
    public ICollection<ShareLink> ShareLinks { get; set; } = new List<ShareLink>();
    
    public double AverageRating => Ratings.Count > 0 ? Ratings.Average(r => r.Value) : 0;
}

File: claude/node/securable/src/models/Note.js
'use strict';

const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Note = sequelize.define(
    'Note',
    {
      id: {
        type: DataTypes.UUID,
        defaultValue: DataTypes.UUIDV4,
        primaryKey: true,
      },
      userId: {
        type: DataTypes.UUID,
        allowNull: false,
        references: { model: 'users', key: 'id' },
      },
      title: {
        type: DataTypes.STRING(255),
        allowNull: false,
        validate: { len: [1, 255], notEmpty: true },
      },
      content: {
        type: DataTypes.TEXT,
        allowNull: false,
        validate: { notEmpty: true },
      },
      isPublic: {
        type: DataTypes.BOOLEAN,
        allowNull: false,
        defaultValue: false,
      },
    },
    {
      tableName: 'notes',
      indexes: [
        { fields: ['user_id'] },
        { fields: ['is_public'] },
      ],
    }
  );

  Note.associate = (models) => {
    Note.belongsTo(models.User, { foreignKey: 'userId', as: 'owner' });
    Note.hasMany(models.Attachment, { foreignKey: 'noteId', as: 'attachments' });
    Note.hasMany(models.Rating, { foreignKey: 'noteId', as: 'ratings' });
    Note.hasMany(models.ShareLink, { foreignKey: 'noteId', as: 'shareLinks' });
  };

  return Note;
};
